### ISSUE TITLE
sys-fs/fuse can't be optimized with any -O level

### ISSUE THREAD

**Issue by thineye** [InBetweenNames/gentooLTO#142]
During my world rebuild sys-fs/encfs failed to build because of undefined reference "__fuse_read_cmd" in the library libfuse.so.

I checked with readelf libfuse.so and with any -O level, except for -O0 (no optimization), it seems that "__fuse_read_cmd" is optimized away.

However if I rebuild sys-fs/fuse with -O0 (but lto/graphite is still enabled), it is not optimized away and sys-fs/encfs build successfully.

**Comment by thineye** [comment]
So I rebuilded my whole system with -fPIC. I don't know if this is an important information, but if I deactivate lto in the flags, "__fuse_read_cmd" is still there.

However only adding *FLAGS+=-ffat-lto-objects, was not enough. I had to do: *FLAGS-=-flto*

**Comment by InBetweenNames** [comment]
Can't reproduce -- `sys-fs/fuse-2.9.7-r1`builds fine here with full optimizations.

Is this on version `3.2.x` by any chance? I see those are masked currently for testing.

I'm unsure if `-fPIC` would affect things, but I know it will at least for sure generate much poorer code than otherwise.  Is there a reason you enabled that flag system wide?  I know some users enable PIC/PIE for security reasons.

**Comment by thineye** [comment]
No, I wanted to rebuild world with -fPIC, but somehow I ereased it in the cflags. However, I had rebuild word also with -fPIC, but still same error.

No, I had stable fuse.

Well, fuse is building well, but the build of sys-fs/encfs is failing because of the undefinded reference.
Is sys-fs/encfs building OK for you?

**Comment by InBetweenNames** [comment]
Yup, I can install `sys-fs/encfs` no problem.

**Comment by InBetweenNames** [comment]
I'm curious, could you share why you have chosen to use `-fPIC` globally on your system?
